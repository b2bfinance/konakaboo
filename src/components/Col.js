import { queries } from '../utils/media';
import styled from 'styled-components';

const width = (query, width) => {
  if (!width) {
    return;
  }

  return queries[query]`
    flex: 0 0 ${width}%;
    max-width: ${width}%;
  `;
};

const Col = styled.div`
  width: 100%;
  min-height: 1px;

  ${props => width('phone', props.phone)};
  ${props => width('tablet', props.tablet)};
  ${props => width('desktop', props.desktop)};
`;

export default Col;
