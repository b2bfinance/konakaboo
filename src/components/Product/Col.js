import React from 'react';
import styled from 'styled-components';
import Col from '../Col';

export const Wrapper = styled(Col)`
  align-items: center;
  background-color: ${props =>
    props.background ? props.theme.productColBackground : ''};
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 1rem 1.25rem;
  text-align: center;

  img {
    flex-shrink: 0;
    max-width: 100%;
    max-height: 120px;
  }

  span {
    font-size: 1.25rem;
    font-weight: ${props => props.theme.mainBoldFontWeight};
    margin-bottom: 0.5rem;
    line-height: 1.1;
  }

  small {
    font-size: 95%;
    font-weight: ${props => props.theme.mainNormalFontWeight};
  }
`;

export const MultiCol = ({ columns, theme }) =>
  columns.map((column, i) => (
    <Wrapper
      key={i}
      phone="100"
      desktop={60 / columns.length}
      background={(i + 1) % 2}
      theme={theme}
    >
      <small>{column.label}</small>
      <span>{column.value}</span>
    </Wrapper>
  ));

export default Wrapper;
