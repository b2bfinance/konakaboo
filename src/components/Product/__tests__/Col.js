import React from 'react';
import { render } from 'enzyme';
import Col, { MultiCol } from '../Col';

describe('Col', () => {
  test('renders correctly', () => {
    expect(render(<Col theme={stubData.theme} />)).toMatchSnapshot();
  });
});

describe('MultiCol', () => {
  test('renders correctly', () => {
    const columns = [
      { label: 'foo', value: 'bar' },
      { label: 'bar', value: 'foo' }
    ];

    expect(
      render(<MultiCol columns={columns} theme={stubData.theme} />)
    ).toMatchSnapshot();
  });
});
