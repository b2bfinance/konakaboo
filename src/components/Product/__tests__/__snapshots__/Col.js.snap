// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Col renders correctly 1`] = `
.c1 {
  width: 100%;
  min-height: 1px;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1rem 1.25rem;
  text-align: center;
}

.c0 img {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  max-width: 100%;
  max-height: 120px;
}

.c0 span {
  font-size: 1.25rem;
  font-weight: 400;
  margin-bottom: 0.5rem;
  line-height: 1.1;
}

.c0 small {
  font-size: 95%;
  font-weight: 300;
}

<div
  class="c0 c1"
>
  
</div>
`;

exports[`MultiCol renders correctly 1`] = `
Array [
  .c1 {
  width: 100%;
  min-height: 1px;
  -webkit-flex: 0 0 100%;
  -ms-flex: 0 0 100%;
  flex: 0 0 100%;
  max-width: 100%;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #f5f6f7;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1rem 1.25rem;
  text-align: center;
}

.c0 img {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  max-width: 100%;
  max-height: 120px;
}

.c0 span {
  font-size: 1.25rem;
  font-weight: 400;
  margin-bottom: 0.5rem;
  line-height: 1.1;
}

.c0 small {
  font-size: 95%;
  font-weight: 300;
}

@media (min-width:62em) {
  .c1 {
    -webkit-flex: 0 0 30%;
    -ms-flex: 0 0 30%;
    flex: 0 0 30%;
    max-width: 30%;
  }
}

<div
    class="c0 c1"
>
    <small>
        foo
    </small>
    <span>
        bar
    </span>
</div>,
  .c1 {
  width: 100%;
  min-height: 1px;
  -webkit-flex: 0 0 100%;
  -ms-flex: 0 0 100%;
  flex: 0 0 100%;
  max-width: 100%;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1rem 1.25rem;
  text-align: center;
}

.c0 img {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  max-width: 100%;
  max-height: 120px;
}

.c0 span {
  font-size: 1.25rem;
  font-weight: 400;
  margin-bottom: 0.5rem;
  line-height: 1.1;
}

.c0 small {
  font-size: 95%;
  font-weight: 300;
}

@media (min-width:62em) {
  .c1 {
    -webkit-flex: 0 0 30%;
    -ms-flex: 0 0 30%;
    flex: 0 0 30%;
    max-width: 30%;
  }
}

<div
    class="c0 c1"
>
    <small>
        bar
    </small>
    <span>
        foo
    </span>
</div>,
]
`;
